src/tutorials/jbearing2.c

```diff
@@ -540,12 +542,18 @@ PetscErrorCode CallPermonAndCompareResults(Tao tao, void *ctx)
   ierr = QPGetSolutionVector(qp, &x_qp);CHKERRQ(ierr);
   ierr = TaoGetSolutionVector(tao,&x_tao);CHKERRQ(ierr);
   
-  /*Different of results from Snes and QP */
+  /* Different of results from Snes and QP */
+  ierr = VecNorm(user->B, NORM_2, &rhs_norm);CHKERRQ(ierr);
+  tao_diff_tol = 1e2 * rtol / rhs_norm;
+  ierr = PetscOptionsGetReal(NULL, NULL, "-tao_diff_tol", &tao_diff_tol, NULL);CHKERRQ(ierr);
+  ierr = PetscOptionsGetBool(NULL, NULL, "-tao_diff_print", &print_tao_diff, NULL);CHKERRQ(ierr);
   ierr = VecCopy(x_tao, x_diff);CHKERRQ(ierr);
   ierr = VecAXPY(x_diff, -1.0, x_qp);CHKERRQ(ierr);
   ierr = VecNorm(x_diff, NORM_2, &x_diff_norm);CHKERRQ(ierr);
-  ierr = PetscPrintf(PETSC_COMM_WORLD,"norm of difference of results from TAO and QP = %e  \n",x_diff_norm);CHKERRQ(ierr);
-  
+  if (x_diff_norm > tao_diff_tol) SETERRQ2(PetscObjectComm((PetscObject)qps), PETSC_ERR_PLIB, "PERMON and TAO yield different results!\nNorm of difference of results from TAO and QP = %e > %e = tolerance",x_diff_norm,tao_diff_tol);
+  if (print_tao_diff) {
```

Comment:
I think we could keep the print by default (we are already printing the tolerances). Moreover, this is an example so getting some feedback is good.

I would suggest getting rid of the option altogether.

Reply:
Fair enough, I will remove it.